Add missing 'values_format' param to disp.plot() in plot_confusion_matrix

blynotes opened the pull request:
I added the missing 'values_format' parameter to the "disp.plot()" call in "plot_confusion_matrix".

<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs
No related issues found.
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.
__Problem Description__
- The class ConfusionMatrixDisplay's plot() function already has the values_format parameter.
- "plot_confusion_matrix" function also has the values_format parameter.
- However, "plot_confusion_matrix" was not passing it's values_format value when calling the ConfusionMatrixDisplay's plot().
- This resulted in being unable to change the format of the values in the confusion matrix when calling plot_confusion_matrix(), even when you passed a value for "values_format". The confusion matrix always displayed in scientific notation.

__My Solution__
- I fixed this by adding the missing parameter to the plot() call in "plot_confusion_matrix".

Simplified example of the code, for further clarification:
```python
   class ConfusionMatrixDisplay:
       ...__init__ function goes here...
       def plot(self, ..., values_format=None, ...):  # <-- values_format is already there.
           ...

   def plot_confusion_matrix(...,
                             values_format=None,  # <-- values_format is already there.
                             ...):
       ...
       disp = ConfusionMatrixDisplay(...)
       return disp.plot(...)  # <-- values_format is missing from here and NOT passed to plot.
```

#### Any other comments?
None. Thank you!

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

blynotes committed:
I added the missing "values_format" parameter to "disp.plot()" call in "plot_confusion_matrix".
"disp" is an object of class ConfusionMatrixDisplay.
ConfusionMatrixDisplay's plot() function already has the values_format parameter.
And the "plot_confusion_matrix" function also has the values_format parameter, but it was not passed when calling plot().

thomasjpfan reviewed (commented):
This needs a test to make sure that `plot_confusion_matrix` passes through `values_format` and changes the text in the plot.

blynotes committed:
Added test coverage to verify 'plot_confusion_matrix' passes through 'values_format' and changes the text in the plot.

blynotes commented:
Hi Thomas, I have updated `test_plot_confusion_matrix.py` with a new test to verify `plot_confusion_matrix` passes through `values_format` and changes the text in the plot.

qinhanmin2014 reviewed (approved):
thanks
Please add an entry to the change log at `doc/whats_new/v*.rst`. Like the other entries there, please reference this pull request with `:pr:` and credit yourself (and other contributors if applicable) with `:user:`.
this should be included in 0.22.1

qinhanmin2014 commented:
ping the reviewers in plot_confusion_matrix PR
@thomasjpfan @NicolasHug @glemaitre @jnothman @amueller

thomasjpfan reviewed (approved):
LGTM! Please add an `BUG` entry to the whats new in 0.22.1.

qinhanmin2014 commented on sklearn/metrics/_plot/tests/test_plot_confusion_matrix.py:
```diff
@@ -231,3 +231,22 @@ def test_confusion_matrix_pipeline(pyplot, clf, data, n_classes):
+    assert disp.text_.shape == (n_classes, n_classes)
+
+    expected_text = np.array([format(v, values_format)
+                              for v in cm.ravel(order="C")])
```
And another small comment, order="C" can be removed here and below.

blynotes replied:
Removed 'order="C"' in both places in the latest commit.  Thanks!

blynotes commented:
Hi Thomas and Hanmin, I have updated the v0.22.rst with a `|Fix|` entry in 0.22.1, and removed `order="C"` per Hanmin's suggestion.

Let me know if there is anything else I need to do.  Thank you!

qinhanmin2014 commented:
Please check the diff @blynotes

blynotes committed:
Merge remote-tracking branch 'upstream/master' into update-plot_confusion_matrix

blynotes committed:
Updated Release Notes and removed 'order="C"' from the cm.ravel() call in test_plot_confusion_matrix.py's test_confusion_matrix_text_format() call.

blynotes commented:
Sorry, I'm new to the `git` commands and still learning.  Please check it now.

qinhanmin2014 committed:
Update v0.22.rst

qinhanmin2014 reviewed (approved):
thanks
